testsuite/test_cim_obj.py

```diff
@@ -2170,541 +2246,650 @@ def test_CIMInstanceName_tocimxmlstr(testcase,
-            "authority without port",
-            'https://10.11.12.13/root/cimv2:CIM_Foo.k1="v1"',
-            dict(
+        None, None, CHECK_0_12_0
+    ),
+    (
+        "authority without port",
+        dict(
+            uri='https://10.11.12.13/root/cimv2:CIM_Foo.k1="v1"',
+            exp_attrs=dict(
                 classname=u'CIM_Foo',
                 keybindings=NocaseDict(k1=u'v1'),
                 namespace=u'root/cimv2',
                 host=u'10.11.12.13'),
-            None, CHECK_0_12_0
         ),
-        (
-            "authority with IPv6 address",
-            'https://[10:11:12:13]/root/cimv2:CIM_Foo.k1="v1"',
-            dict(
+        None, None, CHECK_0_12_0
+    ),
+    (
+        "authority with IPv6 address",
+        dict(
+            uri='https://[10:11:12:13]/root/cimv2:CIM_Foo.k1="v1"',
+            exp_attrs=dict(
                 classname=u'CIM_Foo',
                 keybindings=NocaseDict(k1=u'v1'),
                 namespace=u'root/cimv2',
                 host=u'[10:11:12:13]'),
-            None, CHECK_0_12_0
         ),
-        (
-            "authority with IPv6 address and port",
-            'https://[10:11:12:13]:5989/root/cimv2:CIM_Foo.k1="v1"',
-            dict(
+        None, None, CHECK_0_12_0
+    ),
+    (
+        "authority with IPv6 address and port",
+        dict(
+            uri='https://[10:11:12:13]:5989/root/cimv2:CIM_Foo.k1="v1"',
+            exp_attrs=dict(
                 classname=u'CIM_Foo',
                 keybindings=NocaseDict(k1=u'v1'),
                 namespace=u'root/cimv2',
                 host=u'[10:11:12:13]:5989'),
-            None, CHECK_0_12_0
         ),
-        (
-            "no namespace type",
-            '//10.11.12.13:5989/root/cimv2:CIM_Foo.k1="v1"',
-            dict(
+        None, None, CHECK_0_12_0
+    ),
+    (
+        "no namespace type",
+        dict(
+            uri='//10.11.12.13:5989/root/cimv2:CIM_Foo.k1="v1"',
+            exp_attrs=dict(
                 classname=u'CIM_Foo',
                 keybindings=NocaseDict(k1=u'v1'),
                 namespace=u'root/cimv2',
                 host=u'10.11.12.13:5989'),
-            None, CHECK_0_12_0
         ),
-        (
-            "namespace type http",
-            'http://10.11.12.13:5989/root/cimv2:CIM_Foo.k1="v1"',
-            dict(
+        None, None, CHECK_0_12_0
+    ),
+    (
+        "namespace type http",
+        dict(
+            uri='http://10.11.12.13:5989/root/cimv2:CIM_Foo.k1="v1"',
+            exp_attrs=dict(
                 classname=u'CIM_Foo',
                 keybindings=NocaseDict(k1=u'v1'),
                 namespace=u'root/cimv2',
                 host=u'10.11.12.13:5989'),
-            None, CHECK_0_12_0
         ),
-        (
-            "namespace type upper case HTTP",
-            'HTTP://10.11.12.13:5989/root/cimv2:CIM_Foo.k1="v1"',
-            dict(
+        None, None, CHECK_0_12_0
+    ),
+    (
+        "namespace type upper case HTTP",
```

Comment:
Not sure of nomenclature but should that be scheme  rather than namespace type?  Same below

Reply:
DSP0207 uses the term "namespace type" so I used that too.